src/query/expression/src/types/variant.rs

```diff
@@ -139,3 +141,54 @@ impl ArgType for VariantType {
         StringColumnBuilder::with_capacity(capacity, 0)
     }
 }
+
+pub fn cast_scalar_to_variant(scalar: ScalarRef, buf: &mut Vec<u8>) {
+    let value = match scalar {
+        ScalarRef::Null => common_jsonb::Value::Null,
+        ScalarRef::EmptyArray => common_jsonb::Value::Array(vec![]),
+        ScalarRef::Number(n) => match n {
+            NumberScalar::UInt8(n) => common_jsonb::Value::Number(n.into()),
+            NumberScalar::UInt16(n) => common_jsonb::Value::Number(n.into()),
+            NumberScalar::UInt32(n) => common_jsonb::Value::Number(n.into()),
+            NumberScalar::UInt64(n) => common_jsonb::Value::Number(n.into()),
+            NumberScalar::Int8(n) => common_jsonb::Value::Number(n.into()),
+            NumberScalar::Int16(n) => common_jsonb::Value::Number(n.into()),
+            NumberScalar::Int32(n) => common_jsonb::Value::Number(n.into()),
+            NumberScalar::Int64(n) => common_jsonb::Value::Number(n.into()),
+            // TODO(andylokandy): properly cast Nan and Inf.
+            NumberScalar::Float32(n) => {
+                n.0.try_into()
+                    .map(|n| common_jsonb::Value::Number(n))
+                    .unwrap_or(common_jsonb::Value::Null)
+            }
+            NumberScalar::Float64(n) => {
+                n.0.try_into()
+                    .map(|n| common_jsonb::Value::Number(n))
+                    .unwrap_or(common_jsonb::Value::Null)
+            }
+        },
+        ScalarRef::Boolean(b) => common_jsonb::Value::Bool(b),
+        ScalarRef::String(s) => common_jsonb::Value::String(String::from_utf8_lossy(s)),
+        ScalarRef::Timestamp(Timestamp { ts, .. }) => common_jsonb::Value::Number(ts.into()),
+        ScalarRef::Array(_col) => {
+            // TODO(andylokandy): concantenate bytes of items to array.
+            // let mut items = cast_column_to_variant(col);
+            common_jsonb::Value::Null
+        }
```

Comment:
@b41sh `items` is `StringColumn` of jsonb bytes of items, and we need a way to construct jsonb bytes slice for `Value::Array(items)`.

Reply:
ok, we can add a function to do this